tycho-core/src/main/java/org/eclipse/tycho/core/maven/TychoMavenLifecycleParticipant.java

```diff
@@ -107,6 +111,20 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
         }
     }
 
+    @Override
+    public void afterSessionEnd(MavenSession session) throws MavenExecutionException {
+        if (plexus.hasComponent(EquinoxServiceFactory.class)) {
+            try {
+                EquinoxServiceFactory factory = plexus.lookup(EquinoxServiceFactory.class);
+                if (factory != null) {
```

Comment:
Instead of having a new "refresh" method won't it be an option to check for the Disposable interface her instead?

Reply:
I don't think so because of the different semantics.  That's the problem and I had to change it. Plexus `dispose` method is called automatically by plexus when the bean is _unmanaged_, i.e. removed from the container or when the container is itself destroyed.  
I tried that in the first version but the problem is that once the bean is unmanaged, plexus won't dispose it again.  So the semantics we want is different because we still need the bean to be managed by plexus, but we need to shutdown the container.  I though it would be better to now mix the semantics and rather have a dedicated method (even if `dispose` calls `refresh` at the end.
Now, if you think you'd rather keep a single one, that's fine and it's a matter of adding a check as you indicate.

Reply:
I'm just not sure we should have a "dangerous" method in the public API, if yes we should better call it "shutdown" or something so its clear. But I think it would also be fine to check for disposeable its a very special situation here anyways I think.

Reply:
Or maybe we can check for "Closeable" and let the implementation implement this ?

Reply:
It could also be moved to `EmbeddedEquinox` which makes more sense, or even to a new specific interface.

Reply:
@mickaelistria any preference on how to handle this?

Reply:
And couldn't avoid a new method, and directly call `dispose` here with a comment describing why this needs to be invoked explicitly?
Or if a separate method is really necessary, let's call it `shutdownEquinoxContainer` or `disposeEquinoxContainer` to make it more explicit; and then dispose() would invoke this new method rather than the other way round.

Reply:
> And couldn't avoid a new method, and directly call `dispose` here with a comment describing why this needs to be invoked explicitly?

I think this is enough for now 

```
if (factory instanceof Disposable) {
  ((Dispose)factory).dispose();
}
```